classes/Commands/UpdateCommand.php

```diff
@@ -0,0 +1,200 @@
+        }
+
+        $chainMode = $input->getOption('chain');
+        $noChainMode = $input->getOption('no-chain');
+
+        if ($chainMode && $noChainMode) {
+            $this->logger->error('The chain and no-chain options cannot be active at the same time');
+
+            return ExitCode::FAIL;
+        }
+
+        $this->logger->debug('Starting the update process.');
+
+        try {
+            $prodRootDir = _PS_ROOT_DIR_;
+            $this->logger->debug('Production root directory: ' . $prodRootDir);
+
+            $adminDir = $input->getOption('admin-dir');
+            $this->logger->debug('Admin directory: ' . $adminDir);
+            define('_PS_ADMIN_DIR_', _PS_ROOT_DIR_ . DIRECTORY_SEPARATOR . $adminDir);
+
+            $upgradeContainer = new UpgradeContainer($prodRootDir, $adminDir);
+            $this->logger->debug('Upgrade container initialized.');
+
+            $this->logger->debug('Logger initialized: ' . get_class($this->logger));
+
+            $upgradeContainer->setLogger($this->logger);
+            (new ErrorHandler($this->logger))->enable();
+            $this->logger->debug('Error handler enabled.');
+
+            $configPath = $input->getOption('config-file-path');
+            if (!empty($configPath)) {
+                $exitCode = $this->loadConfiguration($configPath, $upgradeContainer);
+                if ($exitCode !== ExitCode::SUCCESS) {
+                    return $exitCode;
+                }
+                $this->logger->debug('Configuration loaded successfully.');
+            }
+
+            $controller = new AllUpgradeTasks($upgradeContainer);
+            $controller->setOptions([
+            'data' => $input->getOption('data'),
+            'action' => $input->getOption('action'),
+            'channel' => $input->getOption('channel'),
+        ]);
+            $controller->init();
+            $exitCode = $controller->run();
+            $this->logger->debug('Process completed with exit code: ' . $exitCode);
+
+            if ($noChainMode || $exitCode !== ExitCode::SUCCESS) {
+                return $exitCode;
+            }
+
+            return $this->chainCommand($output);
+        } catch (Exception $e) {
+            $this->logger->error('An error occurred during the update process: ' . $e->getMessage());
+
+            return ExitCode::FAIL;
+        }
+    }
+
+    /**
+     * @throws Exception
+     */
+    private function loadConfiguration(string $configPath, UpgradeContainer $upgradeContainer): int
+    {
+        $this->logger->debug('Loading configuration from ' . $configPath);
+        $configFile = file_get_contents($configPath);
+        if (!$configFile) {
+            $this->logger->error('Configuration file not found a location ' . $configPath);
+
+            return ExitCode::FAIL;
+        }
+
+        $inputData = json_decode($configFile, true);
+        $this->logger->debug('Configuration file content: ' . json_encode($inputData));
```

Comment:
What if file contains incorrect JSON?

Reply:
If the parse gives an error, we fall into the console catch

`echo 'An error occurred: '. $e->getMessage() . PHP_EOL;`